evm_loader/ERC20/test/test.py

```diff
@@ -138,383 +44,418 @@ def deployERC20(self, location_hex, location_bin,  mintId, balance_erc20):
                 AccountMeta(pubkey=self.acc.public_key(), is_signer=True, is_writable=False),
                 AccountMeta(pubkey=self.loader.loader_id, is_signer=False, is_writable=False),
                 AccountMeta(pubkey=PublicKey(sysvarclock), is_signer=False, is_writable=False),
             ]))
 
-        result = http_client.send_transaction(trx, self.acc)
-        result = confirm_transaction(http_client, result["result"])
+        result = send_transaction(client, trx, self.acc)
         messages = result["result"]["meta"]["logMessages"]
-        res = messages[messages.index("Program log: succeed") + 1]
-        if not res.startswith("Program log: "):
-            raise Exception("Invalid program logs: no result")
+        res = messages[-1]
+        print("balance_ext", res)
+        if any(search("Program %s failed" % evm_loader_id, m) for m in messages):
+            raise Exception("Invalid program logs: Program %s failed" % evm_loader_id)
         else:
-            return res[13:]
-
-
-    def erc20_mint_id(self, erc20):
+            src_data = result['result']['meta']['innerInstructions'][0]['instructions'][0]['data']
+            data = base58.b58decode(src_data)
+            instruction = data[0]
+            self.assertEqual(instruction, 6)  # 6 means OnReturn
+            self.assertLess(data[1], 0xd0)  # less 0xd0 - success
+            value = data[2:]
+            balance = int.from_bytes(value, "little")
+            print('balance_ext:', balance)
+            return balance
+
+    def erc20_mint_id(self, erc20, erc20_code):
         input = bytearray.fromhex("03e132a122")
         trx = Transaction().add(
             TransactionInstruction(program_id=self.loader.loader_id, data=input, keys=
             [
                 AccountMeta(pubkey=erc20, is_signer=False, is_writable=True),
+                AccountMeta(pubkey=erc20_code, is_signer=False, is_writable=True),
                 AccountMeta(pubkey=self.caller, is_signer=False, is_writable=True),
                 AccountMeta(pubkey=self.acc.public_key(), is_signer=True, is_writable=False),
                 AccountMeta(pubkey=self.loader.loader_id, is_signer=False, is_writable=False),
                 AccountMeta(pubkey=PublicKey(sysvarclock), is_signer=False, is_writable=False),
             ]))
 
-        result = http_client.send_transaction(trx, self.acc)
-        result = confirm_transaction(http_client, result["result"])
+        result = send_transaction(client, trx, self.acc)
         messages = result["result"]["meta"]["logMessages"]
-        res = messages[messages.index("Program log: succeed") + 1]
-        if not res.startswith("Program log: "):
-            raise Exception("Invalid program logs: no result")
+        res = messages[-1]
+        if any(search("Program %s failed" % evm_loader_id, m) for m in messages):
+            raise Exception("Invalid program logs: Program %s failed" % evm_loader_id)
         else:
```

Comment:
this is unnecessary.
please , remove it

Reply:
Fixed